### PR TITLE
[SYCL][Matrix tests] Missing general double type case in initialization and ref compute

### PR DESCRIPTION
**Pull request by dkhaldi**

### CODE REVIEW

File: sycl/test-e2e/Matrix/common.hpp [hunk 1]
```diff
@@ -64,7 +64,9 @@ void matrix_multiply_ref(Ta *A, Tb *B, Tc *C, int M, int N, int K,
             acc += make_fp32(va[i]) * make_fp32(vb[i]);
           else if constexpr (std::is_same_v<Ta, float> &&
                                  std::is_same_v<Tc, float> ||
-                             std::is_integral_v<Ta> && std::is_integral_v<Tc>)
+                             std::is_integral_v<Ta> && std::is_integral_v<Tc> ||
+                             (std::is_same_v<Ta, double> &&
+                              std::is_same_v<Tc, double>))
```
Comment (YuriPlyakhin) [inline sycl/test-e2e/Matrix/common.hpp:69]: nit:
```suggestion
                             std::is_same_v<Ta, double> &&
                              std::is_same_v<Tc, double>)
```

File: sycl/test-e2e/Matrix/common.hpp [hunk 2]
```diff
@@ -127,7 +129,8 @@ void matrix_rand(unsigned int rows, unsigned int cols, T *src, T val) {
 
   for (unsigned int i = 0; i < rows; i++) {
     for (unsigned int j = 0; j < cols; j++) {
-      if constexpr (std::is_same_v<T, bfloat16> || std::is_same_v<T, float>) {
+      if constexpr (std::is_same_v<T, bfloat16> || std::is_same_v<T, float> ||
+                    std::is_same_v<T, double>) {
```
Comment (YuriPlyakhin) [inline sycl/test-e2e/Matrix/common.hpp:133]: question: do we want to use std::uniform_real_distribution<double> for double type?

**Review by YuriPlyakhin — APPROVED** [review]
LGTM. Couple comments - feel free to address or not.

**Comment by dkhaldi** [comment]
@intel/llvm-gatekeepers, can you please help merge?
The failures are independent of this change and have been already reported in https://github.com/intel/llvm/issues/12797 and https://github.com/intel/llvm/issues/12798
Windows Gen12:
Failed Tests (8):
  SYCL :: Assert/assert_in_kernels_win.cpp
  SYCL :: Assert/assert_in_multiple_tus_one_ndebug_win.cpp
  SYCL :: Assert/assert_in_multiple_tus_win.cpp
  SYCL :: Assert/assert_in_one_kernel_win.cpp
  SYCL :: Assert/assert_in_simultaneous_kernels_win.cpp
  SYCL :: Assert/assert_in_simultaneously_multiple_tus.cpp
  SYCL :: Assert/assert_in_simultaneously_multiple_tus_one_ndebug.cpp
  SYCL :: Plugin/sycl-ls-unified-runtime.cpp

**Pull request merged**